src/Symfony/Component/RateLimiter/LimiterInterface.php

```diff
@@ -24,7 +24,7 @@ interface LimiterInterface
      *
      * @param int $tokens the number of tokens required
      */
-    public function consume(int $tokens = 1): bool;
+    public function consume(int $tokens = 1): Limit;
```

Comment:
The `NoLimiter` needs to be updated as well.

Reply:
Yup, I've added a NoLimit object @wouterj wdyt ?

Reply:
I think using the `Limit` object with `$remainingTokens = INF, $retryAfter = now()` would work as well. I think `NoLimiter` is a bit of an edge case - there aren't many use cases for it I think.

Reply:
I don't think there should be a `LimitInterface`. `Limit` should just be constructed in a way that is always "accepted".

Reply:
Nice idea ! I'm a bit tired, thank you all :)